src/archetype.rs

```diff
@@ -43,11 +43,21 @@ pub struct Archetype {
 }
 
 impl Archetype {
-    pub(crate) fn new(types: Vec<TypeInfo>) -> Self {
+    pub(crate) fn empty() -> Self {
+        Self::new_(Vec::new())
+    }
+
+    pub(crate) fn new<B: crate::DynamicBundle>(types: Vec<TypeInfo>) -> Self {
         debug_assert!(
             types.windows(2).all(|x| x[0] < x[1]),
-            "type info unsorted or contains duplicates"
+            "attempted to allocate an entity with duplicate components ({})",
+            core::any::type_name::<B>()
```

Comment:
To clarify, the name of *the duplicated component type* should be printed. The name of the bundle isn't very useful, and is already available in the backtrace.

Reply:
Oh I'm sorry, I misread what you've written before, this of course makes a lot more sense.